Sources/Core/GraphRequestSerializer.swift

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+//
+// You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
+// copy, modify, and distribute this software in source code or binary form for use
+// in connection with the web services and APIs provided by Facebook.
+//
+// As with any software that integrates with the Facebook platform, your use of
+// this software is subject to the Facebook Developer Principles and Policies
+// [http://developers.facebook.com/policy/]. This copyright notice shall be
+// included in all copies or substantial portions of the software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+struct GraphRequestSerializer {
+  let settings: SettingsManaging
+  let logger: Logging
+
+  init(
+    settings: SettingsManaging = Settings.shared,
+    logger: Logging = Logger()
+    ) {
+    self.settings = settings
+    self.logger = logger
+  }
+
+  /**
+   Serializes a graph request and a url
+
+   - Parameters:
+     - url: The url to modify based on the graph request
+     - graphRequest: The request used to provide information to a url
+     - forBatch: Whether or not a graph request is intended to be batched with other requests
+   */
+  func serialize(with url: URL, graphRequest: GraphRequest, forBatch: Bool = false) throws -> URL {
+    if graphRequest.httpMethod == .post, !forBatch {
```

Comment:
Guard requires return. Always prefer a guard statement when returning early.

```suggestion
    guard graphRequest.httpMethod != .post || forBatch else {
```